Add a creator function for Registry that allows a custom Logger

jasobrown opened the pull request:
Add a creator function for `Registry` that allows a custom `Logger`

Also, lower the logging of `Sending measurements` statements to `DEBUG`.

brharrington commented on registry.go:
```diff
@@ -55,11 +55,15 @@ func NewRegistryConfiguredBy(filePath string) (*Registry, error) {
+	return NewRegistryWithLogger(config, defaultLogger())
+}
+
+func NewRegistryWithLogger(config *Config, logger Logger) *Registry {
```
Would it be cleaner to have the logger function and related information come from the config object? It seems like it is just an aspect of how the registry is configured.

jasobrown replied:
Works for me. I'll knock that out shortly ...

jasobrown commented:
... Ignore for now, found a few bugs locally ...

jasobrown commented:
OK, tests passing now. I've moved `Registry.log` to `Config.Log` and fixed up the references. It seemed slightly cleaner to do that than have a `Logger` field in both `Config` and `Registry`; the trade off is needing to dereference `registry.config` every time you want to log something.
